src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -387,20 +389,36 @@ public UUID getFirstSourceDatasetIdFromSnapshotId(
   }
 
   public SnapshotPreviewModel retrievePreview(
-      UUID snapshotId, String tableName, int limit, int offset) {
+      UUID snapshotId,
+      String tableName,
+      int limit,
+      int offset,
+      String sort,
+      SqlSortDirection direction) {
     Snapshot snapshot = retrieve(snapshotId);
 
-    snapshot.getTables().stream()
-        .filter(t -> t.getName().equals(tableName))
-        .findFirst()
-        .orElseThrow(
-            () ->
-                new SnapshotPreviewException(
-                    "No snapshot table exists with the name: " + tableName));
+    SnapshotTable table =
+        snapshot
+            .getTableByName(tableName)
+            .orElseThrow(
+                () ->
+                    new SnapshotPreviewException(
+                        "No snapshot table exists with the name: " + tableName));
+
+    SqlSortDirection sortDirection = Optional.ofNullable(direction).orElse(SqlSortDirection.ASC);
+    if (!StringUtils.equals(sort, PDAO_ROW_ID_COLUMN)) {
```

Comment:
If `sort` can be `null`, maybe that should be an explicit check rather than falling through to the error on line 415. If the null check is done separately, then you no longer need to use `StringUtils.equals()`.

Reply:
That's a good point! I changed this to use the PDAO_ROW_ID_COLUMN as the default so I can just check for equality here.